src/Smidge/SmidgeHelper.cs

```diff
@@ -71,7 +76,7 @@ public class SmidgeHelper : ISmidgeRequire
             _fileBatcher = new FileBatcher(_fileSystem, _requestHelper, hasher);
         }
 
-        public async Task<HtmlString> JsHereAsync(string bundleName, bool debug = false)
+        public async Task<HtmlString> JsHereAsync(string bundleName, bool? debug = null)
```

Comment:
My only concern about changing this to nullable is if this is going to accidentally break anyone usage of these APIs inadvertently if they upgrade since this is changing the signature of this method. Instead, could we declare overloads of this without the debug parameter at all - which would be the one that would use the ISmidgeProfileStrategy ?

Reply:
In some ways i agree that overloads would be preferable, as this means we could mark the debug parameter as deprecated and eventually remove it from the API. 
However, there's a bit of a problem with introducing overloads that don't have the debug parameter - as the current methods with the debug parameter have a default value - meaning the existing method and the new overload become ambiguous. This would then force the caller to have to disambiguate their call.

In testing, I've tried removing the default value from the debug parameter on the existing methods, but arguably this is then changing the existing API too. And in fact an additional problem then arises as some of the methods that take an optional PreProcessPipeline cannot then have a non-optional debug parameter following the optional pipeline. This means the PreProcessPipeline needs to be made non-optional too, which is even more of a breaking change because now callers are forced to specify a PreProcessPipeline value of null even if they are just wanting to specify the debug parameter. 


So, I think I actually favour the current implementation - yes it does alter the API signature slightly, but I don't think this is a breaking change as such. Anyone who was explicitly passing a value for the debug parameter (either true or false) will continue with the current explicit behaviour. 
Anyone who was not explicitly passing a value (and therefore was using the default value False) will now be opted into the new ISmidgeProfileStrategy behaviour. And the default implementation of ISmidgeProfileStrategy is to use the Default profile - which is the equivalent of not using the Debug profile.


Happy to explore alternative approaches if you have any suggestions though.

Reply:
Makes sense. I guess these days binary breaking changes like this don't really matter since everyone builds their solutions when upgrading nuget dependencies - gone are the days of /bin dropping DLLs :P